lib/gvl-tracing.rb

```diff
@@ -28,6 +28,60 @@
 require_relative "gvl_tracing/version"
 
 require "gvl_tracing_native_extension"
+require "json"
 
 module GvlTracing
+  @path = "/tmp/gvl-tracing.json"
+  class << self
+    def start(file)
+      @path = file
+      GvlTracing._start(@path)
+    end
+
+    def stop
+      thread_list = Thread.list
+
+      GvlTracing._stop
+
+      set_thread_name(thread_list)
+    end
+
+    private
+
+    def set_thread_name(list)
+      threads = aggreate_thread_list(list)
+      output_file = File.open(@path)
+      output = output_file.read.split("\n").map do |event|
+        parse_event = JSON.parse(event)
+        thread_id = parse_event.dig("args", "name")
+
+        next(parse_event) unless thread_id
+
+        label = threads[thread_id.strip.to_i]
+
+        next(parse_event) unless label
+
+        parse_event["args"]["name"] = label
+        parse_event
+      end
+      File.write(@path, output.to_json)
+    end
+
+    def aggreate_thread_list(list)
+      list.each_with_object({}) do |t, acc|
+        next unless t.name
```

Comment:
Why skip if it doesn't have a name?

The vast majority of threads aren't named. I think a good thing would be to use the name is available, and if not fallback to parsing its backtrace.

Reply:
The thing that surprise me is that
https://github.com/ivoanjo/gvl-tracing/blob/master/examples/example4.rb#L13
Is properly handled by the C code that look for thread name. But thread list is not able to get it. But in the scenario of the Rails app (like in my initial example), it works properly with `Thread.list`.

That why I skip Thread without name here. I think your proposal is a good idea, but Thread.list needs to report properly thread name like in example 4.

Reply:
I think you provide the answer bellow, and it is also mentioned in [doc](https://rubyapi.org/3.2/o/thread#method-i-name).
> it may set the name to pthread and/or kernel.

Reply:
As a note, example4.rb doesn't work with the new code because the threads die before `.stop` -- thus they won't be in `Thread.list` by the time we call it.

Reply:
Maybe when `t == Thread.main` it's worth printing "Main Thread" or something?

Reply:
Good idea